### DISCUSSION TITLE
VibrationMotor

### DISCUSSION THREAD

**Discussion by Waxime64** [dotnet/Silk.NET#424] [Q&A] [answered]
I try to detect vibration on my gamepad with you lab InputTest. I see in code VibrationMotor didn't load on Silk.NET.Input.Glfw but is loaded on Silk.NET.Input.Sdl Yes.
I change the reference in InputTest: Silk.NET.Input to Silk.NET.Input.Sdl. No error, but my motor speed are always false

I test with xbox controller and with https://gamepad-tester.com/ for start vibration

My new fonction for log vibration in InputTest
`		private static List<IInputDevice> cVibrationDevices = new List<IInputDevice>();
		private static void VibrationMotors() {
			do {
				foreach (var oDevice in cVibrationDevices.ToArray()) {
					if (oDevice is IGamepad oGamepad) {

						foreach (var oMotor in oGamepad.VibrationMotors.ToArray()) {
							if (oMotor.Speed != 0) {
								Console.WriteLine($"GamePad{oDevice.Index}> {oDevice.Name} vibration motor{oMotor.Index}: {oMotor.Speed}");
							}
						}
					}
					System.Threading.Thread.Sleep(100);
				}
			} while (true);
		}`

**Comment by Perksey** [comment, marked as answer]
The vibration motor APIs are for setting a rumble on, not detecting it. Once you've set it to true, it'll return true.
